src/Form/WebdamConfig.php

```diff
@@ -65,8 +104,24 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#title' => $this->t('Client secret'),
       '#default_value' => $config->get('secret'),
       '#description' => $this->t('API Client Secret to use for API access. Contact the Webdam support team to get one assigned. Note that this field will appear blank even if you have previously saved a value.'),
+      '#required' => TRUE,
     ];
 
+    if (!empty($config->get('username')) && !empty($config->get('password'))) {
```

Comment:
Can you do something like `if ($this->isAuthenticated()) {`, and then have that method call `->getSubscriptionDetails()` on the Webdam service? This way, we're not just checking that the username and pw are filled, but that they are valid.